### ISSUE TITLE
GroupBy on an empty array results in an error

### ISSUE THREAD

**Issue by TheMies** [mikhail-barg/jsonata.net.native#9]
First: thanks for this great package!

When I 'group by' data from an empty array, I get an error. This works fine when testing it in the [JSONata Exerciser](https://try.jsonata.org/).

I have this data:
```json
[
  {
    "Id": "a",
    "Issues": [
      {
        "Name": "Foo",
        "Value": "SomeName"
      },
      {
        "Name": "Bar",
        "Value": "SomeOtherName"
      }
    ]
  },
  {
    "Id": "b",
    "Issues": []
  }
]
```

And this is my query:
```
(
    $data := $[Id = "a"].Issues;

    $groupedBy := $data {
        Value : {
            "Count" : $count(Value),
            "Name" : $distinct(Value)
        }
    };
).[*]
```
I get back my results like:
```json
[
  {
    "Count": 1,
    "Name": "SomeName"
  },
  {
    "Count": 1,
    "Name": "SomeOtherName"
  }
]
```
But when filtering on `$data := $[Id = "a"].Issues;` I get an error.
```
Jsonata.Net.Native.JsonataException: T1003: Object key should be String. Expression evaluated to Undefined 'Jsonata.Net.Native.Json.JValue'
   at Jsonata.Net.Native.Eval.EvalProcessor.evalObject(ObjectNode objectNode, JToken input, EvaluationEnvironment env) in /_/src/Jsonata.Net.Native/Eval/EvalProcessor.cs:line 853
```

**Comment by mikhail-barg** [comment]
I was able to make the following repro case:

data:
```json
[
  {
    "Name": "Foo",
    "V": "SomeName"
  }
]
```

query:
```json
${
    V : "z"
}
```

result:
```json
{
  "SomeName": "z"
}
```

and when changing data to
```
[
]
```

error appears instead of an empty object

**Comment by mikhail-barg** [comment]
I've published a new release version 2.0.2 with a fix. Thanks for reporting this!

**Comment by TheMies** [comment]
Thanks for the quick fix!
